domain/src/main/java/org/oppia/android/domain/devoptions/ShowAllHintsAndSolutionHelper.kt

```diff
@@ -0,0 +1,19 @@
+package org.oppia.android.domain.devoptions
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Monitor to identify whether all hints and solution should be shown. */
+@Singleton
+class ShowAllHintsAndSolutionHelper @Inject constructor() {
```

Comment:
This isn't a helper, more like a 'Controller' per our typical structure.

Reply:
Renamed to `ShowAllHintsAndSolutionController`.